tools/clang/tools/dxcompiler/dxcompilerobj.cpp

```diff
@@ -1218,6 +1238,51 @@ class DxcCompiler : public IDxcCompiler3,
         } // PDB in private
       } // Write PDB
 
+      bool updatePartRequired =
+          !opts.PrivateSource.empty() || opts.StripPrivate;
+
+      if (!hasErrorOccurred && updatePartRequired) {
+
+        CComPtr<IDxcContainerBuilder> pContainerBuilder;
+        DxcCreateInstance2(this->m_pMalloc, CLSID_DxcContainerBuilder,
+                           IID_PPV_ARGS(&pContainerBuilder));
```

Comment:
I believe the parts should be added/omitted inside `SerializeDxilContainerForModule`.  Which parts to include in the container is communicated through the `SerializeDxilFlags`.  That will avoid serialization of the container, then rebuilding of the container here, which was part of the point of moving these into the API.  Private Data will also have to be added to the `dxcutil::AssembleInputs` and `SerializeDxilContainerForModule` arguments.  If root signature is supplied from above, it would have to override the one being serialized to pRootSigStream and placed into the container (unless being stripped at the same time).

Reply:
Quick followup: I see you're resetting the root signature in the module when there's an incoming one, so that's ok, and means it doesn't need to be passed down to `SerializeDxilContainerForModule`.  However, we should add the private part while constructing the original container still, rather than modifying it and reserializing it after.

Reply:
Ok yes that makes sense, I had discussed adding the private data in the initial serialization with Xiang but he had recommended I try in a reserialization since there is no validation. But it does makes sense to move the setprivate to avoid the reserialization, I'll go ahead and change that.